pkg/nameresolver/nameresolver_test.go

```diff
@@ -0,0 +1,155 @@
+	cfg := createCfgProviderDefaultNameResolver()
+	propertyExistent := "existent"
+	cfg.Set(propertyExistent, propertyExistent)
+	value, ok := cfg.Get(propertyExistent)
+	assert.Equal(t, propertyExistent, value)
+	assert.True(t, ok)
+}
+
+func TestNameResolverDefault_IsSetPropertyNotFound(t *testing.T) {
+	cfg := createCfgProviderDefaultNameResolver()
+	existent := cfg.IsSet("nonexistent")
+	assert.False(t, existent)
+}
+
+func TestNameResolverDefault_IsSetPropertyFound(t *testing.T) {
+	cfg := createCfgProviderDefaultNameResolver()
+	propertyExistent := "existent"
+	cfg.Set(propertyExistent, propertyExistent)
+	existent := cfg.IsSet(propertyExistent)
+	assert.True(t, existent)
+}
+
+func TestRelaxedName_GetPropertyNotFound(t *testing.T) {
+	cfg := createCfgProviderRelaxedNameResolver()
+	value, ok := cfg.Get("nonexistent")
+	assert.Nil(t, value)
+	assert.False(t, ok)
+}
+
+func TestRelaxedName_GetPropertyFound(t *testing.T) {
+	cfg := createCfgProviderRelaxedNameResolver()
+	propertyExistent := "existent"
+	cfg.Set(propertyExistent, propertyExistent)
+	value, ok := cfg.Get(propertyExistent)
+	assert.Equal(t, propertyExistent, value)
+	assert.True(t, ok)
+}
+
+func TestRelaxedName_IsSetPropertyNotFound(t *testing.T) {
+	cfg := createCfgProviderRelaxedNameResolver()
+	existent := cfg.IsSet("nonexistent")
+	assert.False(t, existent)
+}
+
+func TestRelaxedName_IsSetPropertyFound(t *testing.T) {
+	cfg := createCfgProviderRelaxedNameResolver()
+	propertyExistent := "existent"
+	cfg.Set(propertyExistent, propertyExistent)
+	existent := cfg.IsSet(propertyExistent)
+	assert.True(t, existent)
+}
+
+func createCfgProviderRelaxedNameResolver() provider.Default {
+	cfg := createCfgProviderDefaultNameResolver()
+	cfg.NameResolver = &NameResolverRelaxedName{}
+	return cfg
+}
+
+func TestRelaxedName_GetPropertyFoundEnvFormat(t *testing.T) {
+	cfg := createCfgProviderRelaxedNameResolver()
+	valueExistent := "value"
+	cfg.Set("prop.existent", valueExistent)
+	value, ok := cfg.Get("PROP_EXISTENT")
+	assert.Equal(t, valueExistent, value)
+	assert.True(t, ok)
+}
+
+func TestRelaxedName_GetPropertyFoundYmlFormat(t *testing.T) {
+	cfg := createCfgProviderRelaxedNameResolver()
+	valueExistent := "value"
+	cfg.Set("PROP_EXISTENT", valueExistent)
+	value, ok := cfg.Get("prop.existent")
+	assert.Equal(t, valueExistent, value)
+	assert.True(t, ok)
+}
+
+func TestRelaxedName_IsSetPropertyFoundEnvFormat(t *testing.T) {
+	cfg := createCfgProviderRelaxedNameResolver()
+	cfg.Set("prop.existent", "value")
+	ok := cfg.IsSet("PROP_EXISTENT")
+	assert.True(t, ok)
+}
+
+func TestRelaxedName_IsSetPropertyFoundYmlFormat(t *testing.T) {
+	cfg := createCfgProviderRelaxedNameResolver()
+	cfg.Set("PROP_EXISTENT", "value")
+	ok := cfg.IsSet("prop.existent")
+	assert.True(t, ok)
+}
+
+func createCfgProviderDefaultNameResolver() provider.Default {
+	return provider.Default{
+		Repository:   &repository.Map{},
+		NameResolver: &Default{},
+	}
+}
+
+type NameResolverRelaxedName struct {
```

Comment:
Are you sure we need to test this here?

Reply:
Not necessary. But it's a good case of being tested.